doc/manipulating_images/input_output.rst

```diff
@@ -0,0 +1,258 @@
+.. _extracting_data:
+
+=====================================================
+Input and output: neuroimaging data representation
+=====================================================
+
+.. contents:: **Contents**
+    :local:
+    :depth: 1
+
+|
+
+.. currentmodule:: nilearn.image
+
+.. _loading_data:
+
+Inputing data: file names or image objects
+===========================================
+
+File names and objects, 3D and 4D images
+-----------------------------------------
+
+All Nilearn functions accept file names as arguments::
+
+    >>> from nilearn import image
+    >>> smoothed_img = image.smooth_img('/home/user/t_map001.nii')  # doctest: +SKIP
+
+Nilearn can operate on either file names or `NiftiImage objects
+<http://nipy.org/nibabel/nibabel_images.html>`_. The later represent the
+data loaded in memory. In the example above, the
+function :func:`smooth_img` returns a NiftiImage object, which can then
+be readily passed to other nilearn functions.
+
+In nilearn, we often use the term *"niimg"* as abbreviation that denotes
+either a file name or a `NiftiImage object
+<http://nipy.org/nibabel/nibabel_images.html>`_. 
+
+Niimgs can be 3D or 4D. A 4D niimg may for instance represent a time
+series of 3D images. It can be **a list of file names**, if these contain
+3D information::
+
+    >>> # dataset folder contains subject1.nii and subject2.nii
+    >>> from nilearn.image import smooth_img
+    >>> result_img = smooth_img(['dataset/subject1.nii', 'dataset/subject2.nii']) # doctest: +SKIP
+
+``result_img`` is a 4D in-memory image, containing the data of both
+subjects.
+
+
+.. _filename_matching:
+
+File name matching: "globbing" and user path expansion
+------------------------------------------------------
+
+You can specific files with *wildcard* matching patterns (as in Unix
+shell):
+
+ * **Matching multiple files**: suppose the dataset folder contains
+   subject_01.nii to subject_03.nii ``dataset/subject_*.nii`` is a glob
+   expression matching all filenames::
+
+    >>> # Example with a smoothing process:
+    >>> from nilearn.image import smooth_img
+    >>> result_img = smooth_img("dataset/subject_*.nii") # doctest: +SKIP
+
+   Note that the resulting is a 4D image.
+
+ * **Expanding the home directory** ``~`` is expanded to your home
+   directory::
+
+    >>> result_img = smooth_img("~/dataset/subject_01.nii") # doctest: +SKIP
+
+   Using ``~`` rather than specifying the details of the path is good
+   practice, as it will make it more likely that your script work on
+   different computers.
+
+
+.. topic:: **Python globbing**
+
+    For more complicated use cases, Python also provides functions to work
+    with file paths, in particular, :func:`glob.glob`.
```

Comment:
> Suggestion : create our own glob. It would make life easier for the user (one
> package not to learn)

I disagree. My suggestion: push them to use load_img.

> and we could intellisort the result.

That's a nice idea. We could think about doing that in our sort.